internal/controller/eventing/nats.go

```diff
@@ -29,28 +40,59 @@ func (r *Reconciler) reconcileNATSSubManager(eventing *v1alpha1.Eventing, log *z
 		log.Info("NATS subscription-manager initialized")
 		// save instance only when init is successful.
 		r.natsSubManager = natsSubManager
+	} else {
+		// update the config if hashes differ
+		if eventing.Status.BackendConfigHash != specHash {
+			// set the eventing CR status to processing
+			if err = r.syncStatusWithSubscriptionManagerProcessingWithReason(ctx,
+				eventingv1alpha1.ConditionReasonSubscriptionManagerProcessing,
+				eventing, "Updating NATS subscription-manager with new config.", log); err != nil {
+				return err
+			}
+
+			// stop the subsManager without cleanup
+			if r.isNATSSubManagerStarted {
+				if err := r.stopNATSSubManager(false, log); err != nil {
+					return err
+				}
+				return nil
+			}
+		}
```

Comment:
you may take this out of else block, what do you think? It makes more readable, nested ifs are not so readable.

Reply:
Done